File: coding/15th May 2025/Find Missing and Repeated Values.java
//Find Missing and Repeated Values
class Solution 
{
    public int[] findMissingAndRepeatedValues(int[][] grid) 
    {
        int[] res = new int[2];
        int sum = 0, n = grid.length;
        for(int i = 1; i <= (n * n); i++)
            sum += i;
        HashSet<Integer> set = new HashSet<Integer>();
        for(int i = 0; i < n; i++)
        {
            for(int j = 0; j < n; j++)
            {
                if(set.contains(grid[i][j]))
                    res[0] = grid[i][j];
                else
                {
                    set.add(grid[i][j]);
                    sum -= grid[i][j];
                }
            }
        }
        res[1] = sum;
        return res;
    }
}

File: Coding in Python/30th January 2026/Find Missing and Repeated Values.py
#Find Missing and Repeated Values
class Solution:
    def findMissingAndRepeatedValues(self, grid: List[List[int]]) -> List[int]:
        res = []
        set1 = set()
        for arr in grid:
            for ele in arr:
                if ele in set1:
                    res.append(ele)
                set1.add(ele)
        for i in range(1, len(grid) * len(grid) + 1):
            if i not in set1:
                res.append(i)
        return res
        
